CRM/Core/BAO/ConfigSetting.php

```diff
@@ -216,6 +216,9 @@ public static function applyLocale($settings, $activatedLocales) {
       // and it's valid against our configured locales, we require the session
       // to store this, even if that means starting an anonymous session.
       $session->set('lcMessages', $chosenLocale);
+
+      // Clear menu cache if we've changed locale.
+      CRM_Core_BAO_Navigation::resetNavigation();
```

Comment:
@artfulrobot @mflandorfer I think this might be a separate issue/bug, and I'm it might not be the right fix for when a site has multiple languages enabled (it will flush too often).

The menu has, for example: `civicrm/ajax/navmenu?code=xyz&locale=en_US&cid=204` - so that it can be loaded/cached in multiple languages. Although this more directly affects the browser cache.

Looking at `CRM_Core_BAO_Navigation::getNavigationList()`, the `cacheKeyString` does not include the language, and it probably should (we had that bug in a few places before).

Are there steps to reproduce the issue?

Reply:
@artfulrobot I initially could reproduce the bug on my old sandbox, but not after rebuilding on master.

https://ml1.dev501.symbiodev.xyz/  (admin / `ovQ0fHrzLKtg`)

I also applied this PR, but without the `resetNavigation`.

Other than that, I did some poking around and this PR looks great.